google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Subscriber.java

```diff
@@ -204,19 +202,32 @@ public void close() throws IOException {
   /**
    * Constructs a new {@link Builder}.
    *
-   * <p>Once {@link Builder#build} is called a gRPC stub will be created for use of the {@link
-   * Subscriber}.
-   *
    * @param subscription Cloud Pub/Sub subscription to bind the subscriber to
    * @param receiver an implementation of {@link MessageReceiver} used to process the received
    *     messages
    */
   public static Builder newBuilder(ProjectSubscriptionName subscription, MessageReceiver receiver) {
+    return newBuilder(subscription.toString(), receiver);
+  }
+
+  /**
+   * Constructs a new {@link Builder}.
+   *
+   * @param subscription Cloud Pub/Sub subscription to bind the subscriber to
+   * @param receiver an implementation of {@link MessageReceiver} used to process the received
+   *     messages
+   */
+  public static Builder newBuilder(String subscription, MessageReceiver receiver) {
     return new Builder(subscription, receiver);
   }
 
   /** Subscription which the subscriber is subscribed to. */
   public ProjectSubscriptionName getSubscriptionName() {
+    return ProjectSubscriptionName.parse(subscriptionName);
```

Comment:
Should we also make this a oneof? It isn't right now.

Reply:
I struggled with that myself - I decided not to when I did the refactoring, but I think it would provide better surface consistency.

Reply:
Created https://github.com/GoogleCloudPlatform/google-cloud-java/issues/3021 to track.